### PR TITLE
Add Entry Point To Inject PoS Synthetic Genesis Blocks

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
Current dependencies on/for this PR:
* `feature/pos-syncing-and-steady-state`
  * **PR #925** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/925?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/925?utm_source=stack-comment).

File: lib/pos_consensus.go [hunk 1]
```diff
@@ -49,6 +49,22 @@ func (cc *FastHotStuffConsensus) Start() error {
...
+	}
+
+	// If the consensus is starting for the first time and is at the final PoW block height, then we need
+	// to inject the genesis blocks.
+	if err := cc.tryInjectGenesisBlocks(); err != nil {
+		return errors.Errorf("Error injecting genesis blocks: %v", err)
+	}
+
+	// Fetch the current tip of the chain
+	tipBlock = cc.blockchain.BlockTip()
```
Comment (lazynina) [inline lib/pos_consensus.go:66]: Ahh okay I was confused why we were getting the tipBlock again. Can we add a comment that we fetch the tip again so we have the tip after injecting the PoS genesis blocks?

File: lib/pos_consensus.go [hunk 2]
```diff
@@ -99,6 +115,12 @@ func (cc *FastHotStuffConsensus) HandleLocalBlockProposalEvent(event *consensus.
 		return errors.Errorf("FastHotStuffConsensus.HandleLocalBlockProposalEvent: FastHotStuffEventLoop is not running")
 	}
 
+	// Hold the blockchain's write lock so that the chain cannot be mutated underneath us.
+	// In practice,/ this is a no-op, but it guarantees thread-safety in the event that other
```
Comment (lazynina) [inline lib/pos_consensus.go:119]:
```suggestion
	// In practice, this is a no-op, but it guarantees thread-safety in the event that other
```

File: lib/pos_consensus.go [hunk 3]
```diff
@@ -121,6 +143,12 @@ func (cc *FastHotStuffConsensus) HandleLocalTimeoutBlockProposalEvent(event *con
 		return errors.Errorf("FastHotStuffConsensus.HandleLocalTimeoutBlockProposalEvent: FastHotStuffEventLoop is not running")
 	}
 
+	// Hold the blockchain's write lock so that the chain cannot be mutated underneath us.
+	// In practice,/ this is a no-op, but it guarantees thread-safety in the event that other
```
Comment (lazynina) [inline lib/pos_consensus.go:147]:
```suggestion
	// In practice, this is a no-op, but it guarantees thread-safety in the event that other
```

File: lib/pos_consensus.go [hunk 4]
```diff
@@ -424,6 +452,12 @@ func (cc *FastHotStuffConsensus) HandleBlock(pp *Peer, msg *MsgDeSoBlock) error
 		return errors.Errorf("FastHotStuffConsensus.HandleBlock: FastHotStuffEventLoop is not running")
 	}
 
+	// Hold the blockchain's write lock so that the chain cannot be mutated underneath us.
+	// In practice,/ this is a no-op, but it guarantees thread-safety in the event that other
```
Comment (lazynina) [inline lib/pos_consensus.go:456]:
```suggestion
	// In practice, this is a no-op, but it guarantees thread-safety in the event that other
```

**Review by lazynina — CHANGES_REQUESTED** [review]
All nits on comments

**Pull request closed without merging**